sdk/servicebus/Azure.Messaging.ServiceBus/api/Azure.Messaging.ServiceBus.netstandard2.0.cs

```diff
@@ -430,6 +432,22 @@ public partial class ServiceBusSessionReceiverOptions
         [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
         public override string ToString() { throw null; }
     }
+    public sealed partial class ServiceBusSharedAccessKeyCredential
+    {
+        public ServiceBusSharedAccessKeyCredential(string sharedAccessSignature) { }
+        public ServiceBusSharedAccessKeyCredential(string sharedAccessKeyName, string sharedAccessKey) { }
+        public string SharedAccessKey { get { throw null; } }
+        public string SharedAccessKeyName { get { throw null; } }
+        public string SharedAccessSignature { get { throw null; } }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override string ToString() { throw null; }
+        public void UpdateSharedAccessKey(string keyName, string keyValue) { }
+        public void UpdateSharedAccessSignature(string sharedAccessSignature) { }
```

Comment:
Why not just `Update` to match everyone else?

Reply:
Can you share who everyone else is here? Are you suggesting a single `Update` method instead of two?

Reply:
At one point we had considered a single update method with an overload;  if I recall, there was some concern that callers may be confused and the consensus was that having the name explicitly reflect the intent was more clear.

Reply:
Perhaps @tg-msft is referring to [AzureKeyCredential](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/src/AzureKeyCredential.cs#L54)?

Reply:
There is also StorageSharedKeyCredential, though the method is called [SetAccountKey](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Common/src/StorageSharedKeyCredential.cs#L56).

Reply:
It appears as if `AzureKeyCredential` is intended for API Key scenarios rather than SAS-type scenarios, based on samples that use it.   For example, [Azure.Search](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Azure.Search.Documents/samples/Sample01a_HelloWorld.md)

Reply:
We called it `SetAccountKey` in Storage first and then decided we'd be doing `Update` for all other credentials.  And you have a shared key here so you're doing the same.

Reply:
I don't feel strongly here, though I do tend to favor the explicit naming.   I'm comfortable deferring to @ramya-rao-a , assuming that @tg-msft has cleared this with @KrzysztofCwalina.